nixos/modules/hardware/device-tree.nix

```diff
@@ -142,7 +250,7 @@ in
           default = null;
           example = "*rpi*.dtb";
           description = lib.mdDoc ''
-            Only include .dtb files matching glob expression.
+            Only include .dtb files matching glob expression. Only used with {option}`hardware.deviceTree.dtbSource` option
```

Comment:
Is this restriction needed?

I see that it won't be useful to configure a `filter` when using `dtsSource`, since only one `dtb` will be produced anyway. Here are some reasons to make `filter` work with it anyway:

- A future change could allow multiple `dtb` to be built from sources.
- Conditional application makes options more confusing.
- The default filter is `null` anyway.